src/components/ThemeButton.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { BiSolidMoon, BiSolidSun } from 'react-icons/bi';
+import { Tooltip } from 'react-tooltip';
+
+export default function ThemeButton(): JSX.Element {
+  const preferredTheme = window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light';
+  const selectedTheme = localStorage.theme;
+  const [theme, setTheme] = React.useState<'dark' | 'light'>(selectedTheme || preferredTheme);
```

Comment:
An improvement that could be done would be to set the theme in SWR like we do for /state/saving when saving the book. This would allow us to retrieve this state in ChartJS graph and change the gridlines and colors accordingly. Currently we have static values that may not work properly for light theme (or dark)